x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/entity_descriptions/universal.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EntityDescription } from '../types';
+import { collectValues as collect } from './field_utils';
+
+export const UNIVERSAL_DEFINITION_VERSION = '1.0.0';
+export const UNIVERSAL_IDENTITY_FIELD = 'related.entity';
+
+const entityMetadataExtractorProcessor = {
+  script: {
+    tag: 'entity_metadata_extractor',
+    on_failure: [
+      {
+        set: {
+          field: 'error.message',
+          value:
+            'Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message {{ _ingest.on_failure_message }}',
+        },
+      },
+    ],
+    lang: 'painless',
+    source: `
+Map merged = ctx;
+def id = ctx.entity.id;
+for (meta in ctx.collected.metadata) {
+    Object json = Processors.json(meta);
+    
+    for (entry in ((Map)json)[id].entrySet()) {
```

Comment:
Can we verify if the id exists?

```suggestion
   if (((Map)json)[id] == null) {
     continue;
   }

    for (entry in ((Map)json)[id].entrySet()) {
```